comps/animation/animation.py

```diff
@@ -0,0 +1,220 @@
+
+args = get_args()
+print("args: ", args)
+
+
+# Register the microservice
+@register_microservice(
+    name="opea_service@animation",
+    service_type=ServiceType.ANIMATION,
+    endpoint="/v1/animation",
+    host="0.0.0.0",
+    port=args.port,
+    input_datatype=Base64ByteStrDoc,
+)
+@register_statistics(names=["opea_service@animation"])
+def animate(input: Base64ByteStrDoc):
+    start = time.time()
+
+    if not os.path.exists("inputs"):
+        os.makedirs("inputs")
+    if not os.path.exists("temp"):
+        os.makedirs("temp")
+    if not os.path.exists("outputs"):
+        os.makedirs("outputs")
+
+    print(args.face, args.audio)
+    if not os.path.isfile(args.face):
+        raise ValueError("--face argument must be a valid path to video/image file")
+    elif args.face.split(".")[-1] in ["jpg", "jpeg", "png"]:
+        full_frames = [cv2.imread(args.face)]
+        fps = args.fps
+    else:
+        video_stream = cv2.VideoCapture(args.face)
+        fps = video_stream.get(cv2.CAP_PROP_FPS)
+        print("Reading video frames...")
+        full_frames = []
+        while True:
+            still_reading, frame = video_stream.read()
+            if not still_reading:
+                video_stream.release()
+                break
+            if args.resize_factor > 1:
+                frame = cv2.resize(frame, (frame.shape[1] // args.resize_factor, frame.shape[0] // args.resize_factor))
+            if args.rotate:
+                frame = cv2.rotate(frame, cv2.ROTATE_90_CLOCKWISE())
+
+            y1, y2, x1, x2 = args.crop
+            if x2 == -1:
+                x2 = frame.shape[1]
+            if y2 == -1:
+                y2 = frame.shape[0]
+            frame = frame[y1:y2, x1:x2]
+            full_frames.append(frame)
+
+    print("Number of frames available for inference: " + str(len(full_frames)))
+
+    if args.audio != "None":
+        if not args.audio.endswith(".wav"):
+            os.makedirs("temp", exist_ok=True)
+            print("Extracting raw audio...")
+            # command = f"ffmpeg -y -i {args.audio} -strict -2 temp/temp.wav"
+            # subprocess.call(command, shell=True)
+
+            ffmpeg.input(args.audio).output("temp/temp.wav", strict="-2").run(overwrite_output=True)
+            args.audio = "temp/temp.wav"
+    else:
+        sr, y = base64_to_int16_to_wav(input.byte_str, "temp/temp.wav")
+        args.audio = "temp/temp.wav"
+
+    wav = audio.load_wav(args.audio, 16000)
+    mel = audio.melspectrogram(wav)
+    print("Mel spectrogram shape: " + str(mel.shape))
+    if np.isnan(mel.reshape(-1)).sum() > 0:
+        raise ValueError("Mel contains nan! Using a TTS voice? Add a small epsilon noise to the wav file and try again")
+
+    # one single video frame corresponds to 80/25*0.01 = 0.032 seconds (or 32 milliseconds) of audio
+    # 30 fps video will match closer to audio, than 25 fps
+    mel_chunks = []
+    mel_idx_multiplier = 80.0 / fps
```

Comment:
what does 80 mean here? good to give it a variable or definition instead of magic number